fix: make `Result.value` work with tuple values

zerbina opened the pull request:
Work around a compiler bug by changing the return type of the
`value` template to `untyped`.

The bug in question is: https://github.com/nim-works/nimskull/issues/328

saem commented on lib/experimental/results.nim:
```diff
@@ -200,8 +200,8 @@ func `$`*[E](self: Result[void, E]): string =
 
-template value*[T, E](self: Result[T, E]): T = get self
-template value*[T, E](self: var Result[T, E]): T = get self
+template value*[T, E](self: Result[T, E]): untyped = get self
```
can you leave a comment here to indicate that this should become typed once the bug is fixed?

zerbina replied:
With typed, do you mean `lent T` and `var T`? Or just `T` as was previously?

saem replied:
I meant `T`

zerbina committed:
fix: make `Result.value` work with tuple values

Work around a compiler bug by changing the return type of the
`value` template to `untyped`.

The bug in question is: https://github.com/nim-works/nimskull/issues/328

zerbina commented:
I added a comment regarding the change in the code.

bors r+
